### PR TITLE
feat: Add a Qt object tree view in the debug widget.

### PR DESCRIPTION
**Pull request by iphydf**
This shows the main widget and all its children as a tree.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TokTok/qTox/158)

### CODE REVIEW

File: src/model/debug/debugobjecttreemodel.cpp [hunk 1]
```diff
@@ -0,0 +1,190 @@
+/* SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright © 2024 The TokTok team.
+ */
+
+#include "debugobjecttreemodel.h"
+
+// explicit DebugObjectTreeModel(QObject* parent = nullptr);
```
Comment (robinlinden) [inline src/model/debug/debugobjecttreemodel.cpp:7]: What's up with these comments?

File: src/widget/form/debug/debugobjecttree.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#include "debugobjecttree.h"
```
Comment (robinlinden) [inline src/widget/form/debug/debugobjecttree.cpp:1]: No SPDX in these files?

**Review by robinlinden — APPROVED** [review]

File: src/model/debug/debugobjecttreemodel.cpp (hunk 1 above)
Comment (iphydf) [reply src/model/debug/debugobjecttreemodel.cpp:7]: That was to remind myself of the functions. Deleted now.

File: src/widget/form/debug/debugobjecttree.cpp (hunk 2 above)
Comment (iphydf) [reply src/widget/form/debug/debugobjecttree.cpp:5]: I wish I had cimple to tell me this.

**Pull request merged**